### DISCUSSION TITLE
Actions: support YAML anchors

### DISCUSSION THREAD

**Discussion by hoxu** [community/community#4501] [Actions] [answered]
Please add GitHub Actions YAML anchor support.

(And make popular requests like this visible as issues, so that people can vote and subscribe for them)

**Comment by peaceiris** [comment]
Upvoted!

Another related suggestion on the GitHub Support Community: [Support for YAML anchors - Code to Cloud / GitHub Actions - GitHub Support Community](https://github.community/t/support-for-yaml-anchors/16128)

**Comment by radhikari-arch** [comment]
Upvoted.

**Comment by ptmkenny** [comment]
There's an [issue with even more upvotes here](https://github.com/actions/runner/issues/1182).

**Comment by webertrlz** [comment]
Up

**Comment by jd-sandk** [comment]
We want YAML Anchor Support for GitHub Actions
+1

**Comment by shayki5** [comment]
Any news?

**Reply by ChristopherHX** [reply]
For those who missed that news, if we are lucky then we could have this in spring 2025

> Thanks for engaging on this all, this is on our list to get to in the early part of next year. Right now we are re-working the services in Actions which do our YAML expansion (focusing on how we improve their availability). Once we are live with the new services we will start looking at adding features and this is on our list <3

Source <https://github.com/actions/runner/issues/1182#issuecomment-2317953582>

I didn't expect any movement on this feature as this has been requested for years.

**Comment by janbrasna** [comment, marked as answer]
Aug 4, 2025:
> "YAML anchors are enabled now on github.com" 🎉 
- https://github.com/actions/runner/issues/1182#issuecomment-3150797791 🚀

**Reply by n0099** [reply]
[Dear GitHub: no YAML anchors, please](https://news.ycombinator.com/item?id=45334032)
[GitHub Actions finally supports YAML anchors (but not really)](https://frenck.dev/github-actions-yaml-anchors-aliases-merge-keys/)

**Comment by nebuk89** [comment]
YAML anchors are now shipped :) marking this as closed. I know there are some asks for merge keys, I would ask for a new feedback issue for this so we can track it :) 

As an FYI our logic was to provide what's in the yaml 1.2 spec and merge keys aren't in there. I know this spec is ~15 years old now :) so we are open to the feedback, but want a fresh issue so it's easier to count the 👍 to understand where we put energy next
